proxy/InkAPIInternal.h

```diff
@@ -198,28 +203,18 @@ template <typename ID, int N>
 void
 FeatureAPIHooks<ID, N>::clear()
 {
-  for (int i = 0; i < N; ++i) {
-    m_hooks[i].clear();
-  }
-  hooks_p = false;
-}
-
-template <typename ID, int N>
-void
-FeatureAPIHooks<ID, N>::prepend(ID id, INKContInternal *cont)
-{
-  if (likely(is_valid(id))) {
-    hooks_p = true;
-    m_hooks[id].prepend(cont);
+  for (auto &h : m_hooks) {
+    h.clear();
   }
+  m_hooks_p = false;
 }
 
 template <typename ID, int N>
 void
 FeatureAPIHooks<ID, N>::append(ID id, INKContInternal *cont)
 {
-  if (likely(is_valid(id))) {
-    hooks_p = true;
```

Comment:
Is there a reason to remove the likely hints? At worst, they are pretty harmless right?